### PR TITLE
fix: remove ProofHeight and Proof Validation in msg.ValidateBasic

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

closes: #2536

based on https://github.com/strangelove-ventures/ibc-go/pull/6 by @jtieri

### Commit Message / Changelog Entry

```bash
fix: remove ProofHeight and Proof Validation in msg.ValidateBasic
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/CONTRIBUTING.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/07-tendermint/client_state.go [hunk 1]
```diff
@@ -218,6 +218,15 @@ func (cs ClientState) VerifyMembership(
...
 	value []byte,
 ) error {
+	// check if there is a proof passed in
+	if height.IsZero() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidHeight, "proof height must be non-zero")
+	}
+
+	if len(proof) == 0 {
+		return sdkerrors.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty proof")
+	}
```
Comment (chatton) [inline modules/light-clients/07-tendermint/client_state.go:228]: it looks like we removed test cases which checked for these in ValidateBasic, we should probably add some test cases that cause these errors in the `Verify(Non)MemberShip` functions. WDYT?

**Review by chatton — COMMENTED** [review]
Looks good! Just had one comment about test cases.

File: modules/light-clients/07-tendermint/client_state.go (hunk 1 above)
Comment (AdityaSripal) [reply modules/light-clients/07-tendermint/client_state.go:228]: Agreed

File: modules/light-clients/07-tendermint/client_state_test.go [hunk 2]
```diff
@@ -420,6 +420,12 @@ func (suite *TendermintTestSuite) TestVerifyMembership() {
 				value = []byte("invalid value")
 			}, false,
 		},
+		{
```
Comment (AdityaSripal) [inline modules/light-clients/07-tendermint/client_state_test.go:423]: Can we have the same check for nonmembership?

**Review by AdityaSripal — APPROVED** [review]
Ack, pending minor test additions

File: modules/light-clients/07-tendermint/client_state.go [hunk 3]
```diff
@@ -218,6 +218,15 @@ func (cs ClientState) VerifyMembership(
 	path exported.Path,
 	value []byte,
 ) error {
+	// check if there is a proof passed in
```
Comment (damiannolan) [inline modules/light-clients/07-tendermint/client_state.go:221]: mega nit, but could we remove this comment maybe? I don't think it adds much value

File: modules/light-clients/07-tendermint/client_state.go [hunk 4]
```diff
@@ -264,6 +273,15 @@ func (cs ClientState) VerifyNonMembership(
 	proof []byte,
 	path exported.Path,
 ) error {
+	// check if there is a proof passed in
```
Comment (damiannolan) [inline modules/light-clients/07-tendermint/client_state.go:276]: ditto

**Review by damiannolan — APPROVED** [review]
Nice! Thanks @charleenfei 🚀 

Approved, but agree with above to add same tests for non-membership. Then lgtm :)

**Pull request merged**